runtime/src/snapshot_utils.rs

```diff
@@ -1139,54 +1139,63 @@ pub(crate) fn parse_incremental_snapshot_archive_filename(
     })
 }
 
-/// Get a list of the full snapshot archives in a directory
-pub fn get_full_snapshot_archives<P>(snapshot_archives_dir: P) -> Vec<FullSnapshotArchiveInfo>
-where
-    P: AsRef<Path>,
-{
-    match fs::read_dir(&snapshot_archives_dir) {
-        Err(err) => {
+/// Walk down the snapshot archive directory recursively to collect snapshot archive file info
+fn get_snapshot_archives<T>(dir: &Path, cb: fn(PathBuf) -> Result<T>, output: &mut Vec<T>) {
+    if dir.is_dir() {
+        let entry_iter = fs::read_dir(dir);
+        if let Err(e) = entry_iter {
             info!(
                 "Unable to read snapshot archives directory: err: {}, path: {}",
-                err,
-                snapshot_archives_dir.as_ref().display()
+                e,
+                dir.display()
             );
-            vec![]
+            return;
+        }
+
+        for entry in entry_iter.unwrap() {
```

Comment:
It will look now in every folder and sub-folder of the ledger directory now? Is that right? Does it walk accounts folder? Because we have a lot of files there.

Reply:
good catch. yes, it does. i will limit it to only on `remote` sub directory.